### DISCUSSION TITLE
Postgres query works but same query inside function does not work from supabase dashboard?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#3941] [Questions]
Follwoing query just work fine 
```
SELECT
  subheadings.id as subheading_id,
  headings.paper_id
FROM
  subheadings

LEFT JOIN headings
ON
  subheadings.main_topic_id = headings.id
WHERE subheadings.main_topic_id = 50
```
But same query under rpc shows success but return no rows
```
create or replace function getsyllabusd(paper_idd bigint) 
  returns table(subheading_idd bigint,paper_iddd bigint) 
  language plpgsql
as 
$$
begin
  -- do some work....
  return query
SELECT
  subheadings.id as subheading_id,
 
  headings.paper_id
FROM
  subheadings

LEFT JOIN headings
ON
  subheadings.main_topic_id = headings.id
WHERE subheadings.main_topic_id = paper_idd;

end;
$$ ;

SELECT * FROM getsyllabusd(50);
```

**Comment by silentworks** [comment]
Can you share your `.rpc` code too please?

**Reply by roker15** [reply]
> Can you share your `.rpc` code too please?

I don't have rpc code ,i was testing in supabase sql editor.

**Reply by silentworks** [reply]
In your `returns table` you are stating `subheading_idd` but in your SELECT you are stating `subheading_id`.

**Reply by roker15** [reply]
you can have different select and return name, only type should match

**Comment by steve-chavez** [comment]
> But same query under rpc shows success but return no rows

This is because the dashboard uses a superuser - and it bypases RLS - while rpc doesn't and it follows RLS. Try disabling RLS on your table and check that both dashboard and rpc show the same results.

**Reply by roker15** [reply]
1. my tables headind, subheading... not using rls at all. it's all open ......
2. Both of my queries **simple SELECT** and In **function form** both i am executing from dashboard (sql editor of supabase). I am **not using** **.rpc**. But my simple query return data but same query inside function returns nothing (**but success)**.
3. **Here is surprise** When i call function using rpc id does work and return exptected data.
```
  let { data, error } = await supabase.rpc("getsyllabusd", {
      paper_idd:49,
    });
  
    if (error) console.error(error);
    else console.log(data);
```
**So where is the problem???**

**Reply by steve-chavez** [reply]
> But my simple query return data but same query inside function returns nothing (but success).

Yes, when there's no result, the SQL editor shows `Success. No rows returned`.

> Here is surprise When i call function using rpc id does work and return exptected data.
> paper_idd:49

In your above example you show:

>SELECT * FROM getsyllabusd(50);

Are you sure you get results on rpc as well with `id = 50`? Try `SELECT * FROM getsyllabusd(49)` on the SQL Editor.

At this point I'd suggest isolating the problem and trying a simpler plpgsql function.

**Reply by roker15** [reply]
Yes tested with 49 and 50 both so please explain why this is happening.
